go/vt/vtgate/evalengine/fn_misc.go

```diff
@@ -194,6 +200,33 @@ func (call *builtinInet6Aton) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.VarBinary, Flag: flagNullable, Col: collationBinary}, nil
 }
 
+func (call *builtinLastInsertID) eval(env *ExpressionEnv) (eval, error) {
+	arg, err := call.arg1(env)
+	if err != nil {
+		return nil, err
+	}
+	if arg == nil {
+		env.VCursor().SetLastInsertID(0)
+		return nil, err
+	}
```

Comment:
do we expect this to happen?
I believe we are already checking that argument cannot be nil.

Reply:
no, this is needed. The function accepts and passes through `NULL` values, but changes the session state to 0

```sql
mysql> select last_insert_id(12);
+--------------------+
| last_insert_id(12) |
+--------------------+
|                 12 |
+--------------------+
1 row in set (0.01 sec)

mysql> select last_insert_id(null);
+----------------------+
| last_insert_id(null) |
+----------------------+
|                 NULL |
+----------------------+
1 row in set (0.00 sec)

mysql> select last_insert_id();
+------------------+
| last_insert_id() |
+------------------+
|                0 |
+------------------+
1 row in set (0.00 sec)
```